object_storage_api/services/image.py

```diff
@@ -50,14 +52,19 @@ def create(self, image_metadata: ImagePostMetadataSchema, upload_file: UploadFil
 
         :param image_metadata: Metadata of the image to be created.
         :param upload_file: Upload file of the image to be created.
-        :return: Created image with an pre-signed upload URL.
+        :return: Created image with a pre-signed upload URL.
         :raises InvalidObjectIdError: If the image has any invalid ID's in it.
-        :raises InvalidFilenameExtension: If the image has a mismatched file extension.
+        :raises InvalidFilenameExtension: If the image has a mismatched file extension or the file extension is not
+            supported.
         """
 
-        expected_file_type = mimetypes.guess_type(upload_file.filename)[0]
+        file_extension = Path(upload_file.filename).suffix
+        if not file_extension or file_extension not in config.image.allowed_file_extensions:
+            raise FileTypeMismatchException(f"File extension `{upload_file.filename}` is not supported")
```

Comment:
```suggestion
        file_extension = Path(upload_file.filename).suffix
        if not file_extension or file_extension.toLowerCase() not in config.image.allowed_file_extensions:
            raise FileTypeMismatchException(f"File extension `{upload_file.filename}` is not supported")
```
File extensions can be capitalised sometimes, so you could either do an implementation like this or add the capitalised version of extensions to the allowed list